### DISCUSSION TITLE
Welcome to stryker-net Discussions!

### DISCUSSION THREAD

**Discussion by rouke-broersma** [stryker-mutator/stryker-net#1234] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jidheshr** [comment]
0


I have created a simple Windows WPF Application project (calculator.csproj) in .Net Framework 4.7.2. Also I have created a unit test project (testCalculator.csproj) for the same which is of type Class Library.

When I tried to mutate the calculator project using Stryker command, I am getting the below error :- CSC : error CS5001: Program does not contain a static 'Main' method suitable for an entry point [..\Desktop\calculator\calculator\calculator\calculator.csproj]

The command used is :- ..\Desktop\calculator\calculator\testCalculator>dotnet stryker -s "../calculator.sln"

Does Stryker support WPF application? If so, is there any solution to resolve this error?

**Reply by rouke-broersma** [reply]
Hi @jidheshr 

I don't work with WPF applications so I don't know if this is supposed to work out of the box or not. But I do know that we have had issues with different project types in the past, so it sounds to me like we do not **yet** support WPF. Perhaps you could help make stryker work for WPF? ;) 

I'll create a bug for your question with some info!

**Comment by eranrodrig-tipalti** [comment]
Hi,
when are you planning on releasing the next version?

**Reply by rouke-broersma** [reply]
Hi! Not sure :) We release new versions when we feel it is warranted. Are you waiting on a specific change to be released?

**Comment by ronzhina** [comment]
Hi everyone! 

I have a question about possibility of the Stryker-mutator. Can I run not all tests but only some of them? For example, can I add name of tests class as a parameter for running mutations? 

I want to use the Stryker-mutator in our team, but we have over 13 000 unit tests. And we don't want to run all this tests when we have changes only in one class or method :)

**Reply by rouke-broersma** [reply]
Hi @ronzhina 

This is currently not possible but there is a feature request for this at #420 

13000 unit tests though! Wow that's a lot in one project :) 

You can also use the git diff feature to only run stryker on changed files, we will then also only run the unit tests for those changed files. However in the initial testrun all 13000 tests will unfortunately be run still.

**Reply by ronzhina** [reply]
Big project, a lot of business logic, serious clients :) We have to had an assurance! 

We will be wait this feature! Thank you!

**Comment by psfinaki** [comment]
Hello from Microsoft :)

As mentioned earlier, great project and nice documentation. Could you share the roadmap for implementing the [SolutionRun feature](https://github.com/stryker-mutator/stryker-net/labels/Feat%3A%20Solution%20run)? I found the label and looks like it's in the experimental mode but cannot really figure out how to play with it.

Also, for a workaround, maybe somebody also has some script to go through all the projects in the solution manually and run Stryker on them?

**Reply by rouke-broersma** [reply]
Hi 👋 

To do a solution run, run stryker from the solution folder and pass the solution file name/path (relative) to stryker. A solution run should now be started. It is marked as experimental and not heavily documented because the results are not very consistent and we experience deadlocks on some projects. But, feel free to try it out!

As for a workaround a user has created some powershell scripting to do just that: https://rajbos.github.io/blog/2019/10/04/Runnning-Stryker-in-a-large-solution

**Reply by psfinaki** [reply]
Thanks for the reply!

I am trying things out, will keep you posted then :)

**Reply by psfinaki** [reply]
Yeah well, solution mode is not very consistent yet, doesn't really produce expected output. But the scripts are great, will use them for now.

Anyway, expect more questions from me during upcoming days :)

**Reply by rouke-broersma** [reply]
Happy to answer any questions 👍
